fabric-sound-api-v1/src/client/java/net/fabricmc/fabric/api/client/sound/v1/FabricSoundInstance.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.client.sound.v1;
+
+import java.util.concurrent.CompletableFuture;
+
+import net.minecraft.client.sound.AudioStream;
+import net.minecraft.client.sound.SoundInstance;
+import net.minecraft.client.sound.SoundLoader;
+import net.minecraft.util.Identifier;
+
+/**
+ * General purpose Fabric-provided extensions to {@link SoundInstance}.
+ *
+ * <p>This interface is implicitly implemented on all {@link SoundInstance}s via a mixin and interface injection.
+ */
+public interface FabricSoundInstance {
+	/**
+	 * Load the audio stream for this sound.
+	 *
+	 * <p>By default this will load {@literal .ogg} files from active resource packs. It may be overridden to provide a
+	 * custom {@link AudioStream} implementation which provides audio from another source, such as over the network or
+	 * driven from user input.
+	 *
+	 * @param loader          The default sound loader, capable of loading {@literal .ogg} files.
+	 * @param id              The resolved sound ID, equal to {@link SoundInstance#getSound()}'s location.
```

Comment:
```suggestion
	 * @param id              The resolved sound ID, equal to {@link SoundInstance#getSound()}'s.
```

Reply:
`SoundInstance.getSound()` returns a `Sound`, not an `Identifier`.

Reply:
Well, maybe `'s ID`.

Reply:
I went for `location` because that's the name of the field & getter `Sound`. ID is _fine_, but I don't think it indicates exactly what is being referred to here as well.

Reply:
I think part of the problem here is that it's important to distinguish between the id of the `SoundInstance` (which corresponds to the id in `sounds.json` and `SoundEvent`), and the id of the actual `Sound` being played (which corresponds to a path in a resource pack). Separating the two by having different nomenclature for both feels helpful.